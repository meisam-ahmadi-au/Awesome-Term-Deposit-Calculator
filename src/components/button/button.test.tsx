import { render, screen } from "@testing-library/react";
import Button from "./button.component";
import { describe, it, expect } from "vitest";

describe("Button Component", () => {
  it("should match the snapshot", () => {
    const { asFragment } = render(<Button label="Click Me" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should show with disabled state", () => {
    render(<Button label="Click Me" disabled />);
    const buttonElement = screen.getByText(/click me/i);
    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toBeDisabled();
  });
});
