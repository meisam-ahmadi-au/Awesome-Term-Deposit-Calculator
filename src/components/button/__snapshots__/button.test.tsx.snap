// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Component > should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mb-4 "
  >
    <button
      class="w-full px-4 py-2 rounded-md focus:outline-none bg-blue-500 text-white hover:bg-blue-600"
    >
      Click Me
    </button>
  </div>
</DocumentFragment>
`;
